olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/Types.h

```diff
@@ -48,10 +49,19 @@ class PrefetchTileResult;
 template <typename ResultType>
 using Response = client::ApiResponse<ResultType, client::ApiError>;
 
+/// The union response type that contains the network statistics for the API.
+template <typename ResultType>
+using ExtendedResponse = client::ApiResponse<ResultType, client::ApiError,
+                                             client::NetworkStatistics>;
```

Comment:
I would simply extend the above `Response` with a Payload template parameter defaulted to `void` and make the above response explicit:

```cpp
template <typename ResultType, typename PayloadType = void>
using Response = client::ApiResponse<ResultType, client::ApiError, PayloadType>;

using PartitionsResponse = ExtendedResponse<PartitionsResult, client::NetworkStatistics>;